### ISSUE TITLE
Query timeouts not respected when using regex measurement in query

### ISSUE THREAD

**Issue by rossmcdonald** [influxdata/influxdb#7891]
### Bug report

__System info:__

InfluxDB built from master f126929c3d1462ba9edf2a263dfa4a2ce0c3f61f

__Steps to reproduce:__

1. Set a `[coordinator]` / `query-timeout` in the InfluxDB configuration (for example `"0.01s"`)
2. Execute a query with a regex measurement, for example:

     ```
     select count(*) from _internal."monitor"./.*/
     ```

__Expected behavior:__

The query timeout is applied to the lifetime of the whole query.

__Actual behavior:__

The query timeout seems to be applied to the query for each measurement captured by the regex. For example:

```
> select count(*) from _internal."monitor"./.*/
name,time,count_Alloc,count_Frees,count_HeapAlloc,count_HeapIdle,count_HeapInUse,count_HeapObjects,count_HeapReleased,count_HeapSys,count_Lookups,count_Mallocs,count_NumGC,count_NumGoroutine,count_PauseTotalNs,count_Sys,count_TotalAlloc,count_WALCompactionTimeMs,count_authFail,count_batchesTx,count_batchesTxFail,count_bytesRx,count_cacheAgeMs,count_cacheCompactionDuration,count_cacheCompactionErr,count_cacheCompactions,count_cacheCompactionsActive,count_cachedBytes,count_clientError,count_createFailures,count_currentSegmentDiskBytes,count_diskBytes,count_fieldsCreate,count_memBytes,count_numFiles,count_numMeasurements,count_numMeasurementsDropped,count_numSeries,count_numSeriesDropped,count_oldSegmentsDiskBytes,count_pingReq,count_pointReq,count_pointReqLocal,count_pointsParseFail,count_pointsRx,count_pointsTx,count_pointsWritten,count_pointsWrittenDropped,count_pointsWrittenFail,count_pointsWrittenOK,count_queriesActive,count_queriesExecuted,count_queriesFinished,count_queryDurationNs,count_queryFail,count_queryOk,count_queryReq,count_queryReqDurationNs,count_queryRespBytes,count_readFail,count_req,count_reqActive,count_reqDurationNs,count_seriesCreate,count_serverError,count_snapshotCount,count_statusReq,count_subWriteDrop,count_subWriteOk,count_tsmFullCompactionDuration,count_tsmFullCompactionErr,count_tsmFullCompactions,count_tsmFullCompactionsActive,count_tsmLevel1CompactionDuration,count_tsmLevel1CompactionErr,count_tsmLevel1Compactions,count_tsmLevel1CompactionsActive,count_tsmLevel2CompactionDuration,count_tsmLevel2CompactionErr,count_tsmLevel2Compactions,count_tsmLevel2CompactionsActive,count_tsmLevel3CompactionDuration,count_tsmLevel3CompactionErr,count_tsmLevel3Compactions,count_tsmLevel3CompactionsActive,count_tsmOptimizeCompactionDuration,count_tsmOptimizeCompactionErr,count_tsmOptimizeCompactions,count_tsmOptimizeCompactionsActive,count_writeBytes,count_writeDrop,count_writeDropped,count_writeErr,count_writeError,count_writeFailures,count_writeOk,count_writePointsDropped,count_writePointsErr,count_writePointsOk,count_writeReq,count_writeReqActive,count_writeReqBytes,count_writeReqDurationNs,count_writeReqErr,count_writeReqOk,count_writeTimeout
cq,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,1703,1703,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
ERR: query-timeout limit exceeded
```

A portion of the query succeeded, however it was killed when it reaches a measurement that takes longer than `0.01s` to complete.

/cc @jsternberg

**Comment by rossmcdonald** [comment]
Actually now that I look at this, are partial results returned when the query timeout is exceeded?

**Comment by jsternberg** [comment]
This might be related to the inability to kill queries during the planning phase. Because of the parallel computation of aggregates like `count()`, it finishes computing most of the points usually before query planning finishes. That makes `KILL QUERY` almost useless for those types of queries.

Partial results are returned when the query timeout is exceeded. So it's fully well possible that the query timeout _is_ killing it at the first possible opportunity, but that first possible opportunity is after query planning has already completed.

**Comment by timhallinflux** [comment]
@jsternberg can you confirm that this has been fixed by #8158 ?

**Comment by jsternberg** [comment]
It appears to work when I try it out. I'll close this out.

I tested it by setting the `query-timeout` to 200ms and then running this script. You need to be using the `cli` branch for the experimental client at `github.com/influxdata/influxdb-client`.

```go
package main

import (
	"fmt"
	"math/rand"

	influxdb "github.com/influxdata/influxdb-client"
)

func main() {
	client := influxdb.Client{}
	if err := client.Execute("DROP DATABASE db0; CREATE DATABASE db0"); err != nil {
		panic(err)
	}

	w := client.Writer()
	w.Database = "db0"
	writer := influxdb.NewBufferedWriter(w)
	defer writer.Flush()

	for i := 0; i < 10000; i++ {
		pt := influxdb.Point{
			Name: fmt.Sprintf("cpu_%05d", i),
			Fields: map[string]interface{}{
				"value": rand.Intn(100) + 1,
			},
		}
		pt.WriteTo(writer)
	}

	querier := client.Querier()
	querier.Database = "db0"
	if err := querier.Execute("SELECT count(value) FROM /cpu_.*/"); err != nil {
		fmt.Println(err)
	}
}
```
